src/charm.py

```diff
@@ -99,7 +99,15 @@ def __init__(self, *args):
             source_type="loki",
             source_url=self._external_url,
         )
-        scrape_jobs = [{"static_configs": [{"targets": [f"*:{self._port}"]}]}]
+
+        scrape_url = urlparse(self._external_url)
+        logger.debug("Using url {} for Prometheus scrape".format(urlunparse(scrape_url)))
+        scrape_jobs = [
+            {
+                "metrics_path": "{}/{}".format(scrape_url.path, "metrics"),
```

Comment:
Because trailing slash is not automatically cleaned up
```
>>> "{}/{}".format(urlparse("http://a.b").path, "metrics")
'/metrics'
>>> "{}/{}".format(urlparse("http://a.b/").path, "metrics")
'//metrics'
```
I would
```suggestion
                "metrics_path": "{}/metrics".format(scrape_url.path.rstrip("/")),
```